clients/drcachesim/drpt2trace/drpt2trace.cpp

```diff
@@ -0,0 +1,360 @@
+                          << "." << std::endl;
+                return false;
+            }
+        } else if (strcmp(argv[argidx], "--sb:tsc-offset") == 0) {
+            if (argc <= ++argidx) {
+                std::cerr << CLIENT_NAME << ": --sb:tsc-offset: missing argument."
+                          << std::endl;
+                return false;
+            }
+            uint64_t val = 0;
+            if (sscanf(argv[argidx], "%" PRIx64, &val) > 0) {
+                config.sb_config.tsc_offset = val;
+            } else {
+                std::cerr << CLIENT_NAME
+                          << ": Invalid perf_event_mmap_page.tsc_offset: " << argv[argidx]
+                          << "." << std::endl;
+                return false;
+            }
+        } else if (strcmp(argv[argidx], "--sb:kernel-start") == 0) {
+            if (argc <= ++argidx) {
+                std::cerr << CLIENT_NAME << ": --sb:kernel-start: missing argument."
+                          << std::endl;
+                return false;
+            }
+            uint64_t val = 0;
+            if (sscanf(argv[argidx], "%" PRIx64, &val) > 0) {
+                config.sb_config.kernel_start = val;
+            } else {
+                std::cerr << CLIENT_NAME << ": Invalid kernel start: " << argv[argidx]
+                          << "." << std::endl;
+                return false;
+            }
+        } else if (strcmp(argv[argidx], "--primary-sb") == 0) {
+            if (argc <= ++argidx) {
+                std::cerr << CLIENT_NAME << ": --primary-sb: missing argument."
+                          << std::endl;
+                return false;
+            }
+            config.sb_primary_file_path = std::string(argv[argidx]);
+        } else if (strcmp(argv[argidx], "--secondary-sb") == 0) {
+            if (argc <= ++argidx) {
+                std::cerr << CLIENT_NAME << ": --secondary-sb: missing argument."
+                          << std::endl;
+                return false;
+            }
+            config.sb_secondary_file_path_list.push_back(std::string(argv[argidx]));
+        } else if (strcmp(argv[argidx], "--kcore") == 0) {
+            if (argc <= ++argidx) {
+                std::cerr << CLIENT_NAME << ": --kcore: missing argument." << std::endl;
+                return false;
+            }
+            config.kcore_path = std::string(argv[argidx]);
+        } else {
+            std::cerr << CLIENT_NAME << ": unknown option:" << argv[argidx] << "."
+                      << std::endl;
+            return false;
+        }
+        argidx++;
+    }
+
+    return true;
+}
+
+int
+main(int argc, const char *argv[])
+{
+    options_t options = {};
+    pt2ir_config_t config = {};
+
+    /* Parse the command line.*/
+    if (!process_args(argc, argv, config, options)) {
+        return 1;
+    }
+
+    /* Convert the pt raw data to DR ir */
+    pt2ir_t *ptconverter = new pt2ir_t(config);
```

Comment:
Use a unique_ptr to avoid having to call delete.

Reply:
Done.